httpie/output/streams.py

```diff
@@ -136,7 +137,15 @@ def __init__(
         super().__init__(**kwargs)
         self.formatting = formatting
         self.conversion = conversion
-        self.mime = self.msg.content_type.split(';')[0]
+        self.mime = self.get_mime()
+
+    def get_mime(self) -> str:
+        mime = parse_header_content_type(self.msg.content_type)[0]
+        if self.msg.type_name == 'response':
```

Comment:
I’d much prefer the instance check here. And if this is the only place where we touch `msg.type_name`, then we don’t really need the attribute.